src/main/java/org/terasology/logic/health/DamageAuthoritySystem.java

```diff
@@ -236,4 +236,18 @@ public void onCrash(HorizontalCollisionEvent event, EntityRef entity, CharacterS
             }
         }
     }
+    /**
+     * Reduces the baseDamage of BeforeDamagedEvent if DamageResistComponent is added.
+     *
+     * @param event BeforeDamagedEvent sent before inflicting damage
+     * @param entity Entity which suffered some type of damage
+     */
+    @ReceiveEvent
+    public void onResistMode(BeforeDamagedEvent event, EntityRef entity, DamageResistComponent resistanceComponent) {
+        String damageType=event.getDamageType().getName();
+        String subString=damageType.substring(7);
+        float data=100-resistanceComponent.damageTypes.get(subString);
```

Comment:
`100` here again may be a magic number, I figure you're calculating a percentage - you may also want to consider the impact of using integers and floats together in math (possible loss of precision), Finally you risk errors by assuming several levels of data are all going to be valid and found :-)